design/observers.md

```diff
@@ -0,0 +1,211 @@
+As of now we have Validator Nodes only responsible for write and read requests as well as catching up other Nodes and clients.
+With a state proof support, we can create additional rings of Observer Nodes that can process read requests and reduce load on Validator nodes.
+
+## Goals (in priority order)
+
+- Goal1: Reduce Read Load
+    - Reduce read load on Validators
+
+- Goal2: More Nodes
+    - Have a big number of Nodes available for read requests (we have an essential limit on the size of Validator Nodes because of how chatty RBFT is)
+    
+- Goal3: Reduce Catch-up Load
+    - Reduce catch-up load on Validators
+    
+- Goal4: Reduce Write Load
+    - Be able to send write requests to one Node only (either Validator or Observer)
```

Comment:
Yes, I mean `Gatekeeper or Observer`.
If a client didn't receive a Reply in a timeout from a Gatekeeper/Observer, he assumes that it's malicious and re-sends it with the same reqId. The pool should be able to correctly process requests with the same reqId (it already can do this using a seqno DB) and just Reply immediately if the request was actually ordered.
If a client receives incorrect result from a malicious Observer, then it will be easily detected because of State Proof and BLS multi-signatures for write requests.

As for throttling and DoS, the proposed architecture allows to do this. There is a separate design task to think about this on details (see https://jira.hyperledger.org/browse/INDY-1085).
One of the options is to use Nginx as load balancer and proxy.

Reply:
I will put update Design with these comments.